Oshan uses a transception array instead of shipping launchers

pgmzeta opened the pull request:
<!-- The text between the arrows are comments - they won't be visible on your PR. -->
<!-- To label this PR, add the label(s) without the prefixes surrounded by brackets anywhere, for example: [LABEL] -->
<!-- PRs should at least have one area (A-) label and at least one category (C-) label -->
[station systems][feature]

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Replaces Oshan's cargo intake/outake with transception array, pads, and computers.

Supporting work:
* Add a custom meteorhit effect to the transception array (yeets battery, requires full repair)
* Add repair steps in the transception array's help message
* Add a new define `CARGO_TRANSPCETION`
    * Applied on maps using transception arrays (Nadir, Oshan)
    * Replaces relevant `MAP_OVERRIDE_NADIR` ifdefs
* Slight wording tweak to the modified transception pocket guide

## Why's this needed? <!-- Describe why you think this should be added to the game. -->
Active, 'charged up' hotspots along any portion of the cargo lane means crates disappear into the trench without warning, and often without intent. It's difficult to get the crates back, and repairing the lanes is dangerous with the large number of automatic crate recieves (mail box, special delivery, etc). Crates being launched through water is physics-defying, too :^)

## Screenshots
cargo backroom & array
![image](https://github.com/goonstation/goonstation/assets/91498627/0c1f4823-9179-4936-8c04-3e51f2cfd542)

fullbright / zoomed out
![image](https://github.com/goonstation/goonstation/assets/91498627/05486e23-fb45-46c1-bd65-433966603834)

front office changes include a transception pad and console as well
![image](https://github.com/goonstation/goonstation/assets/91498627/b478542e-64d1-4e02-aca3-ccc26880172e)


## Changelog <!-- If necessary, put your changelog entry below. Otherwise, /please/ delete this entire section. -->
<!-- Put how you want to be credited in the changelog in place of CodeDude. -->
<!-- Use (*) for major changes and (+) for minor changes. See the contributor guide for details. For example: -->

```changelog
(u)glowbold
(*)Oshan utilizes the latest in transception array technology to facilitate cargo needs.
```

TobleroneSwordfish commented:
Pinging @Kubius for thoughts because they made this specifically for Nadir originally

Kubius reviewed (commented):
No big objection here. The array was originally considered a prototype, but there's no reason the tech can't make its way to other suitable locations once it shakes out.

The main thing I'd be mindful of is the site arrangement for the array. The console that reports array status has some control over its function, so it should be somewhere at least mildly secure, and it should have a sizable power cell in the area containing the array itself, since transception requires a fair amount of power.

pgmzeta commented on _std/mapDefines.dm:
```diff
@@ -46,9 +46,11 @@
 
 #elif defined(MAP_OVERRIDE_NADIR)
 #define UNDERWATER_MAP 1
+#define CARGO_TRANSCEPTION 1
```
up on line 49

pgmzeta commented:
> No big objection here. The array was originally considered a prototype, but there's no reason the tech can't make its way to other suitable locations once it shakes out.
> 
> The main thing I'd be mindful of is the site arrangement for the array. The console that reports array status has some control over its function, so it should be somewhere at least mildly secure, and it should have a sizable power cell in the area containing the array itself, since transception requires a fair amount of power.

I matched the 5000 unit power cell for the area APC as with Nadir's transception APC, and the console is in the cargo-access corridor behind cargo. It's a bit exposed by being on the outside, but the windows in the backroom keep it in sight. If sea-meteors cause a consistent 1 hit KO, there may be some more involved shuffling required.

Kubius commented:
One minor supplementary detail is that there's stuff generating on top of the array in the preview and that looks a little bit wonky. Isn't gamebreaking, but could potentially do with adjustment. No-generate fluid turf is one option, the other is tinkering with how the stuff is layered (I'd probably go for the former).

pgmzeta commented:
added nospawn for the transception turfs in the transception array bowl.

also added a custom meteorhit that pops the battery out and requires full repairs, so it getting hit by a meteor is serious but it doesn't qdel.

Kubius commented on code/WorkInProgress/nadir_antenna.dm:
```diff
@@ -298,6 +298,38 @@ TYPEINFO(/obj/machinery/communications_dish/transception)
+		former_cell.throw_at(target,5,1)
+		src.intcap = null
+	src.UpdateIcon()
+	src.visible_message(SPAN_ALERT("<b>[src]'s internal capacitor is violently ejected!</b>"))
```
If I'm reading this correctly, the ejection message will play even if the cell is already ejected. Could have this in the if(src.intcap) block and then have an alternate "is severely damaged" message in an else.

flrsh reviewed (commented):
Apologies for the late review. I am not a fan of transcription arrays and strongly prefer the current shipping system. I have not had the time to look at the code, but I wonder if there's a way to either make the crate path specifically immune to opening up holes to trench or to make objects moving above a certain speed not fall into the trench.

Kubius commented:
Should be entirely possible. Regular pitfalls have a 0.3 second delay before things will fall down them, but it tries the fall wherever the crate ends up being; if the fall check instead retried on the *same* tile after the delay when the falling object is a sellable (crate or artifact), this would mean the object wouldn't fall if it had moved to a different pitfall since that delay.

AXLFAN2000 commented:
Worth noting the other problem this addresses (that I see happen in round a lot more often than the hotspot/trench situation) is crates hitting fish and getting backed up that way.  If this gets denied, some way to prevent fish from stopping cargo crates would be nice too
